python/whylogs/experimental/constraints_generation/types_metrics.py

```diff
@@ -0,0 +1,25 @@
+from typing import List
+
+from whylogs.core.constraints.factories import column_has_non_zero_types
+from whylogs.core.constraints.metric_constraints import MetricConstraint
+from whylogs.core.view.column_profile_view import ColumnProfileView
+
+
+def generate_column_types_constraints(column_name: str, column_profile: ColumnProfileView) -> List[MetricConstraint]:
+    """
+    Generates constraints for the types metrics of a column.
+    Parameters
+    ----------
+    column_name : str
+        Name of the column
+    column_profile : ColumnProfileView
+        Profile of the column
+    """
+    constraints = []
+    types_metric = column_profile.get_metric("types")
+    non_zero_empty_types = []
+    for key, component in vars(types_metric).items():
+        if component.value != 0:
```

Comment:
When the reference profile has a count of 0 for a type, assert that the production data also has 0 of these types.

That way if the data is sparse, we don't cause failure of the constraint (e.g. mixed columns containing strings, integers, fractions, but we only get 2 rows).

Reply:
Maybe have an optional/configurable minimum number of rows required to generate constraints?

Reply:
I meant sparse data in production, like if you had a system monitoring a web service, you might get only a few calls in some time window so the system might be less noisy if we assert 0 for types we haven't seen in the training or reference data, vs assert > 0 for types we have seen.

Reply:
Follow up is fine.